docs/release_guide.md

```diff
@@ -121,3 +123,28 @@ Make an announcement on the [mailing list](https://groups.google.com/g/openservi
 Skip this step if the release is a release candidate (RC).
 
 Open a pull request against the `main` branch making the same version updates as [above](#update-release-branch-with-patches-and-versioning-changes) so the latest release assets are referenced there.
+
+## Make version changes on docs.openservicemesh.io
+
+Associated version numbers need to be updated in the [docs.openservicemesh.io](https://github.com/openservicemesh/osm-docs/) repo.
+
+The files that need to be updated are:
+
+* [Readme](https://github.com/openservicemesh/osm-docs/blob/main/README.md)
+* [Manual Demo](https://github.com/openservicemesh/osm-docs/blob/main/content/docs/getting_started/manual_demo.md)
+* [Health Probes](https://github.com/openservicemesh/osm-docs/blob/main/content/docs/application_health_probes.md)
```

Comment:
The only references to versions I see here are in logs, so I don't think we need to update those every time we cut a release.